vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -330,11 +331,44 @@ func (feeder *clusterStateFeeder) GarbageCollectCheckpoints() {
 // Fetch VPA objects and load them into the cluster state.
 func (feeder *clusterStateFeeder) LoadVPAs() {
 	// List VPA API objects.
-	vpaCRDs, err := feeder.vpaLister.List(labels.Everything())
+	allVpaCRDs, err := feeder.vpaLister.List(labels.Everything())
 	if err != nil {
 		klog.Errorf("Cannot list VPAs. Reason: %+v", err)
 		return
 	}
+
+	// Define a findElement function to check if clusterState.RecommenderName is in the vpaCRD.Spec.RecommenderName
+	findElement := func(slice []string, val string) bool {
+		for _, item := range slice {
+			if item == val {
+				return true
+			}
+		}
+		return false
+	}
+
+	klog.V(3).Infof("Start selecting the vpaCRDs.")
+	var vpaCRDs []*vpa_types.VerticalPodAutoscaler
+	for _, vpaCRD := range allVpaCRDs {
+		currentRecommenderName := feeder.clusterState.RecommenderName
```

Comment:
Could you please explain the approach here?

From the KEP I understood that the alternative recommenders would be run as separate binaries. Do you see these binaries still based on default VPA code (how would this work then)?
If the alternative recommenders would not share code base with default VPA I think the change here can be simplified to only look for 'empty' and "default", without the need of dynamically storing recommenderName in ClusterState.

Reply:
I am a little bit confused here. Is the design target for `algorithm recommender` or  `controller level recommender`?  
I thought we want to have better abstraction on the algorithm and metrics part so that we can still leverage reconcile logic. But if that's the controller level. We should start different recommender runloop at the beginning. 

@kgolab @wangchen615

Reply:
@kgolab , yes, the alternative recommenders are supposed to run as separate binaries. However, I do not want to exclude the possibility that some other recommender is reusing part of the default recommenders' code. Especially for functions like `loadVPAs`, which will apparently be reusable in other recommender binaries.

> Could you please explain the approach here?
> 
> From the KEP I understood that the alternative recommenders would be run as separate binaries. Do you see these binaries still based on default VPA code (how would this work then)?
> If the alternative recommenders would not share code base with default VPA I think the change here can be simplified to only look for 'empty' and "default", without the need of dynamically storing recommenderName in ClusterState.

Reply:
@Jeffwan , The KEP is about supporting `controller level recommender`. We do start different recommender runloop at the beginning by defining `const RecommenderName = "default"` in `pkg/recommender/main.go`

> I am a little bit confused here. Is the design target for `algorithm recommender` or `controller level recommender`?
> I thought we want to have better abstraction on the algorithm and metrics part so that we can still leverage reconcile logic. But if that's the controller level. We should start different recommender runloop at the beginning.
> 
> @kgolab @wangchen615